webbpsf/webbpsf_core.py

```diff
@@ -885,25 +885,15 @@ def _get_telescope_pupil_and_aberrations(self):
                                 "that type: {}".format(type(self.pupil)))
             # ---- apply pupil intensity and OPD to the optical model
 
-            # TODO - more flexibly be smart about if the pupil size works for the LOM or not...
-
-            if 'npix1024' in pupil_transmission:
-                # The linear model is limited to require 1024 pixels right now, so in this case (the default)
-                # we can use that:
-                pupil_optic = opds.OTE_Linear_Model_WSS(
-                    name='{} Entrance Pupil'.format(self.telescope),
-                    transmission=pupil_transmission,
-                    opd=opd_map,
-                    v2v3=self._tel_coords()
-                )
-            else:
-                _log.warning("Nonstandard resolution pupil, so linear model for OTE mirror moves is not supported")
-                pupil_optic = poppy.FITSOpticalElement(
-                    name='{} Entrance Pupil'.format(self.telescope),
-                    transmission=pupil_transmission,
-                    opd=opd_map,
-                    planetype=poppy.poppy_core.PlaneType.pupil
-                )
+            # get npix from pupil_transmission
+            npix = int(pupil_transmission.split('npix')[-1].split('.')[0])
```

Comment:
I think this may need generalization since there are cases in which the pupil_transmission variable may already be a FITS HDUList. We may need to set npix above in the options  block for pupil intensity (lines 863-885), so it can be set differently as appropriate depending on what the input is.